casper/src/main/scala/coop/rchain/casper/dag/BlockDagKeyValueStorage.scala

```diff
@@ -327,4 +363,62 @@ object BlockDagKeyValueStorage {
       stores.deployPool,
       stores.invalidBlocks
     )
+
+  def deployStatus[F[_]: Sync](
+      d: DeployId,
+      deployStore: KeyValueTypedStore[F, DeployId, Signed[DeployData]],
+      deployIndex: KeyValueTypedStore[F, DeployId, BlockHash],
+      invalidBlocksIndex: KeyValueTypedStore[F, DeployId, BlockMetadata],
+      finalizedSet: Set[BlockHash],
+      executedMap: Map[DeployId, Option[String]],
+      expiredSet: Set[DeployId]
+  ): F[String] = {
+    val pooledF   = deployStore.collect { case (dId, _) => dId }.map(_.contains(d))
+    val expiredF  = Sync[F].delay(expiredSet.contains(d))
+    val includedF = deployIndex.collect { case (dId, _) => dId }.map(_.contains(d))
```

Comment:
Use get by key not iterate over all records.

Reply:
I replaced with `contains` directly on KVStore, not sure how I overlooked this API